pkg/cfn/builder/cluster.go

```diff
@@ -0,0 +1,80 @@
+package builder
+
+import (
+	"net"
+
+	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+	gfn "github.com/awslabs/goformation/cloudformation"
+
+	"github.com/weaveworks/eksctl/pkg/eks/api"
+)
+
+const (
+	cfnOutputClusterCertificateAuthorityData = "CertificateAuthorityData"
+	cfnOutputClusterEndpoint                 = "Endpoint"
+	cfnOutputClusterARN                      = "ARN"
+	cfnOutputClusterStackName                = "ClusterStackName"
+)
+
+type clusterResourceSet struct {
+	rs             *resourceSet
+	spec           *api.ClusterConfig
+	subnets        []*gfn.StringIntrinsic
+	securityGroups []*gfn.StringIntrinsic
+}
+
+func NewClusterResourceSet(spec *api.ClusterConfig) *clusterResourceSet {
+	return &clusterResourceSet{
+		rs:   newResourceSet(),
+		spec: spec,
+	}
+}
+
+func (c *clusterResourceSet) AddAllResources() error {
+	c.rs.template.Description = clusterTemplateDescription
+	c.rs.template.Description += clusterTemplateDescriptionDefaultFeatures
+	c.rs.template.Description += templateDescriptionSuffix
+
+	_, globalCIDR, _ := net.ParseCIDR("192.168.0.0/16")
+
+	subnets := map[string]*net.IPNet{}
+	_, subnets[c.spec.AvailabilityZones[0]], _ = net.ParseCIDR("192.168.64.0/18")
+	_, subnets[c.spec.AvailabilityZones[1]], _ = net.ParseCIDR("192.168.128.0/18")
+	_, subnets[c.spec.AvailabilityZones[2]], _ = net.ParseCIDR("192.168.192.0/18")
+
+	c.addResourcesForVPC(globalCIDR, subnets)
```

Comment:
Should `subnets` and `globalCIDR` live inside `addResourcesForVPC`?

Reply:
These ultimately come from above, TBD where exactly, but in the long term – Cluster API config file is the most likely source for these.